### ISSUE TITLE
[EPIC] Zeebe REST API readiness

### ISSUE THREAD

**Issue by koevskinikola** [camunda/zeebe#16167]
## Introduction

This is the third increment of the Zeebe REST API topic. With the completion this epic, the REST API should be fully usable in the C8 stack by supporting Identity authentication and multi-tenancy. With this increment, Tasklist will have an e2e solution for native Zeebe user tasks that they can rely on.

## Task Breakdown

```[tasklist]
### Gateway API
- [ ] https://github.com/camunda/zeebe/issues/16831
- [ ] https://github.com/camunda/zeebe/issues/16835
- [ ] Ensure that the required resources for the REST API are ready before it becomes available
```

```[tasklist]
### Java client
- [ ] https://github.com/camunda/zeebe/issues/16885
- [ ] https://github.com/camunda/zeebe/issues/16836
- [ ] https://github.com/camunda/zeebe/issues/16837
- [ ] https://github.com/camunda/zeebe/issues/16643
- [ ] https://github.com/camunda/zeebe/issues/16677
```

## Other Teams

- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

## Links

- [Product Hub (internal)
](https://github.com/camunda/product-hub/issues/1926)
- [Cross-team collaboration doc (internal)](https://docs.google.com/document/d/1IxHvRBNa6VtQN53QjbHxfckD8mJqYxEuh5Uvza4pq80/edit)
- [Zeebe breakdown document (internal)](https://docs.google.com/document/d/1qN9gJHtgwli5n7GBqU-mm1GngS7-pQ38-i3v7_ivQVI/edit#heading=h.eqd411rw2pi9)

## Additional context

Blocked by https://github.com/camunda/zeebe/issues/16166

**Comment by npepinpe** [comment]
Missing items are postponed to the next iteration.
